### PR TITLE
Ju 598 java 21 migration fearghus

### PR DESCRIPTION
**Pull request by foman-ch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -414,10 +423,10 @@
                     <dependency>
                         <groupId>uk.gov.companieshouse</groupId>
                         <artifactId>java-checkstyle-config</artifactId>
-                        <version>1.1</version>
+                        <version>${java-checkstyle-config.version}</version>
                     </dependency>
                 </dependencies>
-            </plugin>
+            </plugin>-->
```
Comment (dgurjar-ch) [inline pom.xml]: remove commented code for clean code

File: src/itest/java/uk/gov/companieshouse/disqualifiedofficers/delta/matcher/DisqualificationRequestMatcher.java [hunk 2]
```diff
@@ -15,9 +15,9 @@
  */
 public class DisqualificationRequestMatcher implements ValueMatcher<Request> {
 
-    private String expectedOutput;
-    private String type;
-    private Logger logger;
+    private final String expectedOutput;
+    private final String type;
```
Comment (dgurjar-ch) [inline src/itest/java/uk/gov/companieshouse/disqualifiedofficers/delta/matcher/DisqualificationRequestMatcher.java:19]: Any reason to remove final ?

File: pom.xml (hunk 1 above)
Comment (foman-ch) [reply pom.xml]: resolved

File: pom.xml [hunk 3]
```diff
@@ -224,18 +227,24 @@
...
+                            <exclusions>
+                                <exclusion>
+                                    <groupId>org.xmlresolver</groupId>
+                                    <artifactId>xmlresolver</artifactId>
+                                </exclusion>
+                            </exclusions>
                         </dependency>
                     </dependencies>
-                </plugin>
+                </plugin>-->
```
[the remark this answers is not in the document]
Comment (foman-ch) [reply pom.xml]: resolved :)

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -28,3 +28,10 @@ wiremock:
   server:
     port: 8888
 
+management:
+  endpoints:
+    web:
```
Comment (dgurjar-ch) [inline src/main/resources/application.yml:33]: Other healthCheck parameters are missing  like enabled-by-default=false ,health.show-details=never

**Review by dgurjar-ch — APPROVED** [review]

File: src/main/resources/application.yml (hunk 4 above)
Comment (foman-ch) [reply src/main/resources/application.yml:33]: added additional parameters now

**Review by jaddison-ch — APPROVED** [review]

**Pull request merged**